<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualization of clustifyr results â€¢ clustifyrdata</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visualization of clustifyr results">
<meta property="og:description" content="clustifyrdata">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">clustifyrdata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/benchmark.html">Benchmarking clustifyr</a>
    </li>
    <li>
      <a href="../articles/download_refs.html">Prebuilt references for clustifyr</a>
    </li>
    <li>
      <a href="../articles/otherformats.html">Support for single cell object formats</a>
    </li>
    <li>
      <a href="../articles/othertools.html">Other considerations w/ clustifyr</a>
    </li>
    <li>
      <a href="../articles/references.html">Reference building for clustifyr</a>
    </li>
    <li>
      <a href="../articles/visualizations.html">Visualization of clustifyr results</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/rnabioco/clustifyrdata/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="visualizations_files/header-attrs-2.1/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Visualization of clustifyr results</h1>
            
      
      <small class="dont-index">Source: <a href="http://github.com/rnabioco/clustifyrdata/blob/master/vignettes/visualizations.Rmd"><code>vignettes/visualizations.Rmd</code></a></small>
      <div class="hidden name"><code>visualizations.Rmd</code></div>

    </div>

    
    
<div id="plotting-tsne-and-umap-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-tsne-and-umap-results" class="anchor"></a>Plotting tSNE and UMAP results</h2>
<p><code>clustifyr</code> provides several functions to plot tSNE or UMAP results. The <code><a href="http://rnabioco.github.io/clustifyr/reference/plot_dims.html">plot_dims()</a></code> function will plot tSNE or UMAP data using a meta.data table and can color the cells based on cluster identity.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">clustifyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">clustifyrdata</span>)

<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tibble</span>)

<span class="co"># Matrix of normalized single-cell RNA-seq counts</span>
<span class="no">pbmc_matrix</span>[<span class="fl">1</span>:<span class="fl">3</span>, <span class="fl">1</span>:<span class="fl">3</span>]
<span class="co">#&gt; 3 x 3 sparse Matrix of class "dgCMatrix"</span>
<span class="co">#&gt;               AAACATACAACCAC AAACATTGAGCTAC AAACATTGATCAGC</span>
<span class="co">#&gt; AL627309.1                 .              .              .</span>
<span class="co">#&gt; AP006222.2                 .              .              .</span>
<span class="co">#&gt; RP11-206L10.2              .              .              .</span>

<span class="co"># meta.data table containing cluster assignments for each cell</span>
<span class="no">pbmc_meta</span>[<span class="fl">1</span>:<span class="fl">5</span>, ]
<span class="co">#&gt;                orig.ident nCount_RNA nFeature_RNA percent.mt RNA_snn_res.0.5</span>
<span class="co">#&gt; AAACATACAACCAC     pbmc3k       2419          779  3.0177759               1</span>
<span class="co">#&gt; AAACATTGAGCTAC     pbmc3k       4903         1352  3.7935958               3</span>
<span class="co">#&gt; AAACATTGATCAGC     pbmc3k       3147         1129  0.8897363               1</span>
<span class="co">#&gt; AAACCGTGCTTCCG     pbmc3k       2639          960  1.7430845               2</span>
<span class="co">#&gt; AAACCGTGTATGCG     pbmc3k        980          521  1.2244898               6</span>
<span class="co">#&gt;                seurat_clusters   classified     UMAP_1     UMAP_2</span>
<span class="co">#&gt; AAACATACAACCAC               1 Memory CD4 T   2.341260 -3.0761487</span>
<span class="co">#&gt; AAACATTGAGCTAC               3            B   3.712070 11.7883301</span>
<span class="co">#&gt; AAACATTGATCAGC               1 Memory CD4 T   5.065264 -0.7839952</span>
<span class="co">#&gt; AAACCGTGCTTCCG               2   CD14+ Mono -11.783583  0.8343185</span>
<span class="co">#&gt; AAACCGTGTATGCG               6           NK   3.576498 -8.9778881</span>

<span class="co"># Create tSNE and color cells based on cluster ID</span>
<span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/plot_dims.html">plot_dims</a></span>(
  <span class="kw">x</span> <span class="kw">=</span> <span class="st">"UMAP_1"</span>, <span class="co"># name of column in the meta.data containing the data to plot on x-axis</span>
  <span class="kw">y</span> <span class="kw">=</span> <span class="st">"UMAP_2"</span>, <span class="co"># name of column in the meta.data containing the data to plot on y-axis</span>
  <span class="kw">data</span> <span class="kw">=</span> <span class="no">pbmc_meta</span>, <span class="co"># meta.data table containing cluster assignments for each cell</span>
  <span class="kw">feature</span> <span class="kw">=</span> <span class="st">"seurat_clusters"</span> <span class="co"># name of column in meta.data to color cells by</span>
)</pre></body></html></div>
<p><img src="visualizations_files/figure-html/Create%20tSNE-1.png" width="960" style="display: block; margin: auto;"></p>
<p>Cells can also be colored based on the expression level of a gene or list of genes using the <code><a href="http://rnabioco.github.io/clustifyr/reference/plot_gene.html">plot_gene()</a></code> function.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co"># Create tSNE and color cells based on gene expression</span>
<span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/plot_gene.html">plot_gene</a></span>(
  <span class="kw">x</span> <span class="kw">=</span> <span class="st">"UMAP_1"</span>, <span class="co"># name of column in the meta.data containing the data to plot on x-axis</span>
  <span class="kw">y</span> <span class="kw">=</span> <span class="st">"UMAP_2"</span>, <span class="co"># name of column in the meta.data containing the data to plot on y-axis</span>
  <span class="kw">expr_mat</span> <span class="kw">=</span> <span class="no">pbmc_matrix</span>, <span class="co"># matrix of normalized single-cell RNA-seq counts</span>
  <span class="kw">metadata</span> <span class="kw">=</span> <span class="no">pbmc_meta</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span>(<span class="st">"rn"</span>), <span class="co"># meta.data table containing cluster assignments for each cell</span>
  <span class="kw">genes</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CD79B"</span>, <span class="st">"CD8A"</span>), <span class="co"># vector of gene names to color cells</span>
  <span class="kw">cell_col</span> <span class="kw">=</span> <span class="st">"rn"</span> <span class="co"># name of column in meta.data containing the cell IDs</span>
)
<span class="co">#&gt; [[1]]</span></pre></body></html></div>
<p><img src="visualizations_files/figure-html/Overlay%20gene%20expression-1.png" width="700" style="display: block; margin: auto;"></p>
<pre><code>#&gt; 
#&gt; [[2]]</code></pre>
<p><img src="visualizations_files/figure-html/Overlay%20gene%20expression-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="visualizing-clustifyr-results" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizing-clustifyr-results" class="anchor"></a>Visualizing <code>clustifyr()</code> results</h2>
<p>The <code>clustifyr()</code> function outputs a matrix of correlation coefficients and <code><a href="http://rnabioco.github.io/clustifyr/reference/clustify_lists.html">clustify_lists()</a></code> and <code><a href="http://rnabioco.github.io/clustifyr/reference/clustify_nudge.html">clustify_nudge()</a></code> output positive scores. <code>clustifyr</code> provides built-in functions to help visualize these results.</p>
<p>Cell type assignments can be assessed by plotting the <code>clustifyr()</code> correlation matrix as a heatmap using the <code><a href="http://rnabioco.github.io/clustifyr/reference/plot_cor_heatmap.html">plot_cor_heatmap()</a></code> function.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co"># Run clustifyr()</span>
<span class="no">res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/clustify.html">clustify</a></span>(
  <span class="kw">input</span> <span class="kw">=</span> <span class="no">pbmc_matrix</span>, <span class="co"># matrix of normalized single-cell RNA-seq counts</span>
  <span class="kw">metadata</span> <span class="kw">=</span> <span class="no">pbmc_meta</span>, <span class="co"># meta.data table containing cluster assignments for each cell</span>
  <span class="kw">ref_mat</span> <span class="kw">=</span> <span class="no">cbmc_ref</span>, <span class="co"># reference matrix containing bulk RNA-seq data for each cell type</span>
  <span class="kw">query_genes</span> <span class="kw">=</span> <span class="no">pbmc_vargenes</span>, <span class="co"># list of highly varible genes identified with Seurat</span>
  <span class="kw">cluster_col</span> <span class="kw">=</span> <span class="st">"seurat_clusters"</span> <span class="co"># name of column in meta.data containing cell clusters</span>
)

<span class="co"># Create heatmap using the clustifyr correlation matrix</span>
<span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/plot_cor_heatmap.html">plot_cor_heatmap</a></span>(
  <span class="kw">cor_mat</span> <span class="kw">=</span> <span class="no">res</span> <span class="co"># matrix of correlation coefficients from clustifyr()</span>
)</pre></body></html></div>
<p><img src="visualizations_files/figure-html/Correlation%20heatmap-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The <code><a href="http://rnabioco.github.io/clustifyr/reference/plot_cor.html">plot_cor()</a></code> function can also be used to create a tSNE for each cell type of interest and color the cells based on the correlation coefficients.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co"># Create a tSNE for each cell type of interest and color cells based on correlation coefficients</span>
<span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/plot_cor.html">plot_cor</a></span>(
  <span class="kw">x</span> <span class="kw">=</span> <span class="st">"UMAP_1"</span>, <span class="co"># name of column in the meta.data containing the data to plot on x-axis</span>
  <span class="kw">y</span> <span class="kw">=</span> <span class="st">"UMAP_2"</span>, <span class="co"># name of column in the meta.data containing the data to plot on y-axis</span>
  <span class="kw">cor_mat</span> <span class="kw">=</span> <span class="no">res</span>, <span class="co"># matrix of correlation coefficients from clustifyr()</span>
  <span class="kw">metadata</span> <span class="kw">=</span> <span class="no">pbmc_meta</span>, <span class="co"># meta.data table containing cluster assignments for each cell</span>
  <span class="kw">data_to_plot</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">res</span>)[<span class="fl">1</span>:<span class="fl">2</span>], <span class="co"># name of cell type(s) to plot correlation coefficients</span>
  <span class="kw">cluster_col</span> <span class="kw">=</span> <span class="st">"seurat_clusters"</span> <span class="co"># name of column in meta.data containing cell clusters</span>
)
<span class="co">#&gt; [[1]]</span></pre></body></html></div>
<p><img src="visualizations_files/figure-html/Overlay%20corr%20coefficients-1.png" width="1152" style="display: block; margin: auto;"></p>
<pre><code>#&gt; 
#&gt; [[2]]</code></pre>
<p><img src="visualizations_files/figure-html/Overlay%20corr%20coefficients-2.png" width="1152" style="display: block; margin: auto;"></p>
<p>Cell clusters can also be labeled using the <code><a href="http://rnabioco.github.io/clustifyr/reference/plot_best_call.html">plot_best_call()</a></code> function, which takes the correlation matrix and labels cell clusters with the cell type that has the highest correlation coefficient.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co"># Create tSNE and label clusters with the cell type that has the highest correlation coefficient</span>
<span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/plot_best_call.html">plot_best_call</a></span>(
  <span class="kw">cor_mat</span> <span class="kw">=</span> <span class="no">res</span>, <span class="co"># matrix of correlation coefficients from clustifyr()</span>
  <span class="kw">metadata</span> <span class="kw">=</span> <span class="no">pbmc_meta</span>, <span class="co"># meta.data table containing UMAP or tSNE data</span>
  <span class="kw">do_label</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="co"># should the feature label be shown on each cluster?</span>
  <span class="kw">do_legend</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="co"># should the legend be shown?</span>
  <span class="kw">cluster_col</span> <span class="kw">=</span> <span class="st">"seurat_clusters"</span>
)</pre></body></html></div>
<p><img src="visualizations_files/figure-html/Label%20clusters%20with%20cell%20types-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="assessing-clustifyr-accuracy" class="section level2">
<h2 class="hasAnchor">
<a href="#assessing-clustifyr-accuracy" class="anchor"></a>Assessing <code>clustifyr()</code> accuracy</h2>
<p>The <code>clustifyr()</code> results can also be evaluated by over-clustering the data and comparing the cell type assignments before and after over-clustering. This is accomplished using the <code><a href="http://rnabioco.github.io/clustifyr/reference/overcluster_test.html">overcluster_test()</a></code> function. The cell type assignments should be similar with and without over-clustering.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="co"># Overcluster cells and compare cell type assignments with and without over-clustering</span>
<span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/overcluster_test.html">overcluster_test</a></span>(
  <span class="kw">expr</span> <span class="kw">=</span> <span class="no">pbmc_matrix</span>, <span class="co"># matrix of normalized single-cell RNA-seq counts</span>
  <span class="kw">metadata</span> <span class="kw">=</span> <span class="no">pbmc_meta</span>, <span class="co"># meta.data table containing UMAP or tSNE data</span>
  <span class="kw">ref_mat</span> <span class="kw">=</span> <span class="no">cbmc_ref</span>, <span class="co"># reference matrix containing bulk RNA-seq data for each cell type</span>
  <span class="kw">cluster_col</span> <span class="kw">=</span> <span class="st">"seurat_clusters"</span>, <span class="co"># name of column in meta.data containing cell clusters</span>
  <span class="kw">n</span> <span class="kw">=</span> <span class="fl">5</span> <span class="co"># expand cluster number n-fold for overclustering</span>
)</pre></body></html></div>
<p><img src="visualizations_files/figure-html/Overclustering%20test-1.png" width="960" style="display: block; margin: auto;"></p>
<p>The cell types from the bulk RNA-seq reference matrix can also be mixed together using the <code>make_comb_ref()</code> function to assess the specificity of the cell type assignments. If a cluster shows a higher correlation when using the mixed reference matrix, this suggests that the cluster contains multiple cell types.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co"># Create reference containing different combinations of the bulk RNA-seq data</span>
<span class="no">comb_ref</span> <span class="kw">&lt;-</span> <span class="fu">make_comb_ref</span>(
  <span class="kw">ref_mat</span> <span class="kw">=</span> <span class="no">cbmc_ref</span> <span class="co"># reference matrix containing bulk RNA-seq data for each cell type</span>
)

<span class="co"># Peek at the new combined reference</span>
<span class="no">comb_ref</span>[<span class="fl">1</span>:<span class="fl">5</span>, <span class="fl">1</span>:<span class="fl">5</span>]

<span class="co"># Run clustifyr() using the combined reference</span>
<span class="no">comb_res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/clustify.html">clustify</a></span>(
  <span class="kw">input</span> <span class="kw">=</span> <span class="no">pbmc_matrix</span>, <span class="co"># matrix of normalized single-cell RNA-seq counts</span>
  <span class="kw">metadata</span> <span class="kw">=</span> <span class="no">pbmc_meta</span>, <span class="co"># meta.data table containing cluster assignments for each cell</span>
  <span class="kw">ref_mat</span> <span class="kw">=</span> <span class="no">comb_ref</span>, <span class="co"># reference matrix containing bulk RNA-seq data for each cell type</span>
  <span class="kw">query_genes</span> <span class="kw">=</span> <span class="no">pbmc_vargenes</span>, <span class="co"># list of highly varible genes identified with Seurat</span>
  <span class="kw">cluster_col</span> <span class="kw">=</span> <span class="st">"seurat_clusters"</span> <span class="co"># name of column in meta.data containing cell clusters</span>
)

<span class="co"># Create tSNE and label clusters with the assigned cell types from the combined reference</span>
<span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/plot_best_call.html">plot_best_call</a></span>(
  <span class="kw">cor_mat</span> <span class="kw">=</span> <span class="no">comb_res</span>, <span class="co"># matrix of correlation coefficients from clustifyr()</span>
  <span class="kw">metadata</span> <span class="kw">=</span> <span class="no">pbmc_meta</span>, <span class="co"># meta.data table containing UMAP or tSNE data</span>
  <span class="kw">do_label</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="co"># should the feature label be shown on each cluster?</span>
  <span class="kw">do_legend</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="co"># should the legend be shown?</span>
  <span class="kw">cluster_col</span> <span class="kw">=</span> <span class="st">"seurat_clusters"</span>
)</pre></body></html></div>
</div>
<div id="plotting-other-attributes-from-the-meta-data-table" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-other-attributes-from-the-meta-data-table" class="anchor"></a>Plotting other attributes from the meta.data table</h2>
<p>Visualization of other attributes shared in the metadata between ref and query by <code>plot_cols</code>, such as nGene, nUMI, mt_percentage, as another way of identity confirmation after <code>clustify</code>. Certain cell types have distinct patterns, more genes detected, for example.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Rui Fu, Jay Hesselberth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.0.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
