<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Support for single cell object formats â€¢ clustifyrdata</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Support for single cell object formats">
<meta property="og:description" content="clustifyrdata">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">clustifyrdata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/benchmark.html">Benchmarking clustifyr</a>
    </li>
    <li>
      <a href="../articles/download_refs.html">Prebuilt references for clustifyr</a>
    </li>
    <li>
      <a href="../articles/geo-annotations.html">Improving NCBI GEO submissions of scRNA-seq data</a>
    </li>
    <li>
      <a href="../articles/otherformats.html">Support for single cell object formats</a>
    </li>
    <li>
      <a href="../articles/othertools.html">Other considerations w/ clustifyr</a>
    </li>
    <li>
      <a href="../articles/references.html">Reference building for clustifyr</a>
    </li>
    <li>
      <a href="../articles/visualizations.html">Visualization of clustifyr results</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/rnabioco/clustifyrdata/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="otherformats_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Support for single cell object formats</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/rnabioco/clustifyrdata/blob/master/vignettes/otherformats.Rmd"><code>vignettes/otherformats.Rmd</code></a></small>
      <div class="hidden name"><code>otherformats.Rmd</code></div>

    </div>

    
    
<p><code>clustifyr</code> is aimed for general use and flexibility in various analysis pipelines. While its core code functions with matrices and dataframes, and hence has minimal package dependencies, wrapper functions are provided for further convenience of the user.</p>
<div id="direct-handling-of-seurat-objects-v2-and-v3-extracting-the-required-data-and-inserting-called-cell-types-directly-into-the-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#direct-handling-of-seurat-objects-v2-and-v3-extracting-the-required-data-and-inserting-called-cell-types-directly-into-the-objects" class="anchor"></a>Direct handling of <code>seurat</code> objects, v2 and v3, extracting the required data, and inserting called cell types directly into the objects</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://github.com/rnabioco/clustifyr#readme">clustifyr</a></span>)

<span class="co"># can even returning a new object with `type` added to metadata.</span>
<span class="kw">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/clustify.html">clustify</a></span>(
  input = <span class="kw">s_small</span>,
  cluster_col = <span class="st">"res.1"</span>,
  ref_mat = <span class="kw">cbmc_ref</span>,
  seurat_out = <span class="fl">TRUE</span>
)

<span class="co"># or return a correlation matrix, if seurat_out is set to F</span>
<span class="kw">res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/clustify_lists.html">clustify_lists</a></span>(
  input = <span class="kw">s_small</span>,
  marker = <span class="kw">pbmc_markers</span>,
  marker_inmatrix = <span class="fl">FALSE</span>,
  cluster_col = <span class="st">"res.1"</span>,
  seurat_out = <span class="fl">FALSE</span>
)
</pre></div>
</div>
<div id="other-scrna-seq-object-formats-are-easily-supported-by-adding-defaults-to-a-lookup-file-object_loc_lookup-currently-including-sce-urd-cds-fsce-" class="section level2">
<h2 class="hasAnchor">
<a href="#other-scrna-seq-object-formats-are-easily-supported-by-adding-defaults-to-a-lookup-file-object_loc_lookup-currently-including-sce-urd-cds-fsce-" class="anchor"></a>Other scRNA-seq object formats are easily supported, by adding defaults to a lookup file <code>object_loc_lookup</code>, currently including <code>SCE</code>, <code>URD</code>, <code>CDS</code>, <code>FSCE</code>.</h2>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co"># location of data in object in this format</span>
<span class="kw">object_loc_lookup</span>
<span class="co">#&gt;              SingleCellExperiment               URD</span>
<span class="co">#&gt; expr  input@assays$data$logcounts input@logupx.data</span>
<span class="co">#&gt; meta as.data.frame(input@colData)        input@meta</span>
<span class="co">#&gt; var                          NULL   input@var.genes</span>
<span class="co">#&gt; col                    cell_type1           cluster</span>
<span class="co">#&gt;                         FunctionalSingleCellExperiment</span>
<span class="co">#&gt; expr input@ExperimentList$rnaseq@assays$data$logcounts</span>
<span class="co">#&gt; meta               input@ExperimentList$rnaseq@colData</span>
<span class="co">#&gt; var                                               NULL</span>
<span class="co">#&gt; col                                     leiden_cluster</span>
<span class="co">#&gt;                             Seurat</span>
<span class="co">#&gt; expr         input@assays$RNA@data</span>
<span class="co">#&gt; meta               input@meta.data</span>
<span class="co">#&gt; var  input@assays$RNA@var.features</span>
<span class="co">#&gt; col                  RNA_snn_res.1</span>
<span class="co">#&gt;                                                                                                                CellDataSet</span>
<span class="co">#&gt; expr do.call(function(x) {row.names(x) &lt;- input@featureData@data$gene_short_name; return(x)}, list(input@assayData$exprs))</span>
<span class="co">#&gt; meta                                                                                   as.data.frame(input@phenoData@data)</span>
<span class="co">#&gt; var                     as.character(input@featureData@data$gene_short_name[input@featureData@data$use_for_ordering == T])</span>
<span class="co">#&gt; col                                                                                                           Main_Cluster</span>
<span class="co">#&gt;                                                                                                                                                                                                                                                                                              H5File</span>
<span class="co">#&gt; expr                                                                                                                {mat &lt;- input[["layers/norm_data"]][,];\ncolnames(mat) &lt;- input[["row_attrs/gene_names"]][];\nrownames(mat) &lt;- input[["col_attrs/cell_names"]][];\nmat &lt;- t(mat);\nreturn(mat)}</span>
<span class="co">#&gt; meta {l &lt;- input$ls(recursive=TRUE);\ns &lt;- l[,1][stringr::str_detect(l[,1], "col_attrs/")];\ns2 &lt;- sapply(s, FUN=function(x) {input[[x]][]});\ncolnames(s2) &lt;- stringr::str_remove(colnames(s2),"col_attrs/");\nmeta &lt;- tibble::column_to_rownames(as.data.frame(s2), "cell_names");\nreturn(meta)}</span>
<span class="co">#&gt; var                                                                                                                                                                                                                                                                                            NULL</span>
<span class="co">#&gt; col                                                                                                                                                                                                                                                                                      orig.ident</span>

<span class="co"># adding seurat3 locations</span>
<span class="kw">object_loc_lookup</span><span class="op">$</span><span class="kw">Seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
  expr = <span class="st">"input@assays$RNA@data"</span>,
  meta = <span class="st">"input@meta.data"</span>,
  var = <span class="st">"input@assays$RNA@var.features"</span>,
  col = <span class="st">"RNA_snn_res.1"</span>
)
</pre></div>
<p>For example, sample code for <code>scrunchy</code> is as easy as:</p>
<pre class='{"example"}'><code>res &lt;- clustify_nudge(
  input = fsce_small,
  ref_mat = cbmc_ref,
  marker = cbmc_m,
  query_genes = pbmc_vargenes,
  cluster_col = "k_cluster"
)</code></pre>
</div>
<div id="making-new-references-from-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#making-new-references-from-objects" class="anchor"></a>Making new references from objects</h2>
<p>Making scRNA-seq objects into references is assisted by wrappers as well.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># for both v2 and v3</span>
<span class="kw">seurat_ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/object_ref.html">object_ref</a></span>(
  input = <span class="kw">s_small3</span>,
  cluster_col = <span class="st">"RNA_snn_res.1"</span>
)

<span class="co"># and other object types with lookup</span>
<span class="kw">object_ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://rnabioco.github.io/clustifyr/reference/object_ref.html">object_ref</a></span>(
  input = <span class="kw">sce_small</span>,
  cluster_col = <span class="st">"cell_type1"</span>
)
</pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Rui Fu, Jay Hesselberth.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
